### ISSUE TITLE
PHP8.1 issue

### ISSUE THREAD

**Issue by 3kbest** [php-curl-class/php-curl-class#690]
```
During inheritance of ArrayAccess: Uncaught ErrorException: Return type of Curl\CaseInsensitiveArray::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/xxx/vendor/php-curl-class/php-curl-class/src/Curl/CaseInsensitiveArray.php:92
Stack trace:
#0 /var/www/xxx/vendor/php-curl-class/php-curl-class/src/Curl/CaseInsensitiveArray.php(5): 
```

**Comment by zachborboa** [comment]
Referencing your pull request here #691.

**Comment by zachborboa** [comment]
Fixed in #691.
